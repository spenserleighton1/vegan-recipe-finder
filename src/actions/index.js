export const addRecipes = (recipes) => ({
  type: 'ADD_RECIPES',
  recipes
})

export const searchByIngredient = (ingredients) => ({
  type: 'SEARCH_BY_INGREDIENT', 
  ingredients
})