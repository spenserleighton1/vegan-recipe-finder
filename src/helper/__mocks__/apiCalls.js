export const fetchRecipes = jest.fn().mockImplementation(() => ({
  recipes: [{title: 'a food'}, {title: 'more a food'}, {title: 'even more a food'}]
}));