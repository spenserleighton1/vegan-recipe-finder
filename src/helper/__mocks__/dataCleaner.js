export const cleanData = jest.fn();

export const cleanRecipe = jest.fn();