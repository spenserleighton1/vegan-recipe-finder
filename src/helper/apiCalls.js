export const fetchRecipes = async (key) => {
  const url = `https://food2fork.com/api/search?key=${key}&q=vegan`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
}