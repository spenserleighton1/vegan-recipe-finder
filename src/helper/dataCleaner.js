export const cleanData = data => {
  const cleanData = data.map(recipe => {
    return {     
      recipeTitle: recipe.title,
      recipeLink: recipe.source_url,
      recipeImage: recipe.image_url,
      recipeId: recipe.recipe_id }
    })
  return cleanData;
}

export const cleanRecipe = data => ({
  title: data.recipe.title,
  publisher: data.recipe.publisher,
  ingredients: data.recipe.ingredients,
  linkUrl: data.recipe.source_url,
  id: data.recipe.recipe_id 
})
