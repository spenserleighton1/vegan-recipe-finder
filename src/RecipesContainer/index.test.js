import React from 'react';
import { RecipesContainer, mapDispatchToProps, mapStateToProps } from './';
import { shallow, mount } from 'enzyme';
import { addSingleRecipe, isLoading } from '../actions';
import { fetchSingleRecipe } from '../helper/apiCalls';
import { cleanRecipe} from '../helper/dataCleaner';

jest.mock('../helper/apiCalls');
jest.mock('../helper/dataCleaner');

describe('RecipesContainer', () => {
  it('should match the snapshot', () => {
    const mockRecipes = [{ title: 'recipe 1' }, { title: 'recipe 2' }]
    const wrapper = shallow(<RecipesContainer recipes={ mockRecipes }/>);

    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke addSingleRecipe when fetch recipe is called', async () => {
    const mockRecipes = [{ title: 'recipe 1' }, { title: 'recipe 2' }]
    const mockAddSingleRecipe = jest.fn();
    const mockIsLoading = jest.fn();
    const wrapper = shallow(<RecipesContainer isLoading={ mockIsLoading } addSingleRecipe={ mockAddSingleRecipe } recipes={ mockRecipes }/>);

    const results = await wrapper.instance().fetchRecipe()

    expect(mockAddSingleRecipe).toHaveBeenCalled()
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with addSingleRecipe', () => {
      const mockAddSingleRecipe = jest.fn();
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = addSingleRecipe({ title: 'a recipe' });

      mappedProps.addSingleRecipe({ title: 'a recipe' });
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with isLoading', () => {
      const mockIsLoading = jest.fn();
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = isLoading({ loading: false });

      mappedProps.isLoading({ loading: false });
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object', () => {
      const mockState = {recipes: []};
      const expected = {"recipes": []};

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})