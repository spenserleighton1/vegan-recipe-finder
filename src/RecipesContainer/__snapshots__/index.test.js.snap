// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipesContainer
    recipes={
        Array [
            Object {
              "title": "recipe 1",
            },
            Object {
              "title": "recipe 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RecipeCard
          fetchRecipe={[Function]}
          title="recipe 1"
/>,
        <RecipeCard
          fetchRecipe={[Function]}
          title="recipe 2"
/>,
      ],
      "className": "recipes-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "fetchRecipe": [Function],
          "title": "recipe 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "fetchRecipe": [Function],
          "title": "recipe 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RecipeCard
            fetchRecipe={[Function]}
            title="recipe 1"
/>,
          <RecipeCard
            fetchRecipe={[Function]}
            title="recipe 2"
/>,
        ],
        "className": "recipes-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "fetchRecipe": [Function],
            "title": "recipe 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "fetchRecipe": [Function],
            "title": "recipe 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
